src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String READ_SCRIPT
+            = "SELECT * FROM books WHERE id = ?";
+    private static final String CREATE_SCRIPT
+            = "INSERT INTO books(id, title, price) VALUES (?, ?, ?)";
+    private static final String UPDATE_SCRIPT
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_SCRIPT
+            = "DELETE FROM books WHERE id = ?";
+    private static final String ALL_SCRIPT
+            = "SELECT * FROM books";
+    private static final String FIND_MAX_ID_SCRIPT
+            = "SELECT MAX(id) AS id FROM books";
+
+    @Override
+    public Book create(Book book) {
+        book.setId(findMaxID() + 1);
+        try (PreparedStatement statement = ConnectionUtil.getConnection()
+                .prepareStatement(CREATE_SCRIPT)) {
+            statement.setLong(1, book.getId());
+            statement.setString(2, book.getTitle());
+            statement.setBigDecimal(3, book.getPrice());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a new book", e);
+        }
+        return book;
```

Comment:
Move inside try block
```suggestion
        return book;
        } catch (SQLException e) {
            throw new DataProcessingException("Can't create a new book", e);
        }
```